### PR TITLE
fix(chutes): land B3-stream review fixes missed by #789's stuck PR-sync

### PR DESCRIPTION
**Pull request by Evrard-Nil**
Follow-up to **#789** (stream-side `<think>`→`reasoning_content`). #789 merged, but **GitHub's PR-branch sync stalled at the first commit `b01f3f6a`** and never picked up the two follow-up commits, so the merge landed the stale tree **without the review correctness fixes**. Verified: `compare 2c9fff04...main` → `diverged, behind_by=2`.

This **cherry-picks `2c9fff04`** (the reviewed correctness fix) onto current main. Two fixes:
1. **Preserve held-back pending bytes** when the parser latches off because a native-reasoning (`reasoning_content`/`reasoning`) delta arrives *while still scanning* a held whitespace prefix — without this, those held bytes are dropped (**content-loss** in that edge case). Adds regression test `native_reasoning_after_held_whitespace_does_not_lose_content`.
2. Make `apply_think_to_event` **atomic** — chunk and `raw_bytes` never de-sync if a re-parse fails.

Net effect: main now matches what reviewers actually approved on the #789 branch (`e9cae1a7`), minus the no-op CI-nudge commit.

`cargo test -p inference_providers --lib attested::chutes` → 106 pass; clippy/fmt clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by lloydmak99 — APPROVED** [review]
Approve. Both fixes are correct:

Fix 1 (native-reasoning pending loss): when a native-reasoning delta arrives while Scanning, the held-back whitespace/partial-<think> bytes are now prepended to the current delta's content before latching to Done. Previously those bytes were orphaned (flush() is a no-op in Done). The regression test pins this exactly.

Fix 2 (atomicity): apply_think_to_event now returns the event fully untouched if the ChatCompletionChunk re-parse fails, rather than rewriting raw_bytes while leaving chunk stale. This is the same atomicity pattern as rewrite_sse_event_model and closes the concern I noted in the #789 review.

File: crates/inference_providers/src/attested/chutes/mod.rs [hunk 1]
```diff
@@ -992,6 +992,21 @@ impl ThinkStreamState {
...
+            if !self.pending.is_empty() {
+                let held = std::mem::take(&mut self.pending);
+                let rest = delta
+                    .get("content")
+                    .and_then(Value::as_str)
+                    .unwrap_or_default();
+                delta.insert(
+                    "content".to_string(),
+                    Value::String(format!("{held}{rest}")),
+                );
```
[the remark this answers is not in the document]
Comment (Evrard-Nil) [reply crates/inference_providers/src/attested/chutes/mod.rs]: Good catch — applied in 0ebc5922. Reused the owned `held` String via `push_str(rest)` instead of `format!`, dropping the extra allocation in the stream hot path. Covered by the existing `native_reasoning_after_held_whitespace_does_not_lose_content` regression test (106 chutes tests still green, clippy/fmt clean).

**Review by PierreLeGuen — APPROVED** [review]
This cleanly cherry-picks the two already-reviewed `ThinkStreamState` fixes from #789 that were lost to the stuck PR-branch sync: prepending held `pending` bytes to the native-reasoning delta before latching to `Done` (previously dropped, since `flush()` is a no-op in `Done`), and the early return in `apply_think_to_event` that keeps `chunk`/`raw_bytes` in sync on re-parse failure. Traced both against the surrounding parser logic — the prepend path can't clobber non-string content (`ChatDelta.content` is `Option<String>`, models.rs:25), and the regression test covers the exact content-loss scenario. No blocking issues.

Checks: `cargo test -p inference_providers --lib attested::chutes` (106 passed, includes the new `native_reasoning_after_held_whitespace_does_not_lose_content` regression test), `cargo fmt --check`, and `cargo clippy -p inference_providers --lib` — all clean locally; CI rollup (`Lint`, `Test Suite`, `security_audit`) was still in progress on the latest head `0ebc592`.

**Review by lloydmak99 — APPROVED** [review]
Re-approve. Latest commit is a trivial allocation optimization (push_str instead of format! to prepend pending bytes onto delta content) — no behavior change, existing test covers it.
